microservices-connector/internal/controller/gmconnector_controller.go

```diff
@@ -144,7 +124,7 @@ func reconcileResource(ctx context.Context, dynamicClient *dynamic.DynamicClient
 	}
 
 	var resources []string
-	appliedCfg, err := applyCustomConfig(step, svcCfg, yamlFile)
+	appliedCfg, err := applyRouterConfig(step, svcCfg, yamlFile)
```

Comment:
we can move the router part separately instead of repeating the processing for each step's `recondileResource`

Reply:
this is not repeating, if the step is not router, if will not go on, this happens the code walking through every step of the gragh, router is one of them, if we make it outside, as a seperate step, we will need to process the graph twice.

Reply:
Can we rename it back to `applyCustomConfig` since it is not a router specific change? though for other step we just convert to string

Reply:
how about patchCustomConfig? since it does not really apply the config to k8s, just change the content